org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/text/TMPresentationReconciler.java

```diff
@@ -420,7 +422,21 @@ private void themeChange(ITokenProvider oldTheme, ITokenProvider newTheme, IDocu
 	@Override
 	public void install(ITextViewer viewer) {
 		Assert.isNotNull(viewer);
-
+		
+		if (this.grammar == null && viewer.getDocument() != null) {
+			try {
+				ContentTypeInfo info = ContentTypeHelper.findContentTypes(viewer.getDocument());
+				if (info != null) {
+					this.grammar = findGrammar(info);
+				}
+			} catch (CoreException e) {
+				TMUIPlugin.getDefault().getLog().log(new Status(IStatus.ERROR, TMUIPlugin.PLUGIN_ID, e.getMessage(), e));
+			}
+			if (this.grammar == null) {
+				throw new IllegalArgumentException("No TextMate Grammar for provided editor/viewer/document.");

```

Comment:
With Oxygen, `viewer.getDocument() != null` always return `false`, so this block is never executed and the exception never thrown